src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -151,8 +165,12 @@ public static object CreateEnumerableValue(ref Utf8JsonReader reader, ref ReadSt
                 return null;
             }
 
-            Type propType = state.Current.JsonPropertyInfo.RuntimePropertyType;
-            if (typeof(IList).IsAssignableFrom(propType))
+            Type propertyType = state.Current.JsonPropertyInfo.RuntimePropertyType;
+            if (JsonClassInfo.IsNonGenericInterface(propertyType))
```

Comment:
Not sure why we can't prevent this logic by using a RuntimePropertyType of `List<T>` - initialized when the JsonPropertyInfo is.

Reply:
i.e. when we see a collection interface when creating the JsonPropertyInfo - whether non generic or generic - we should just be able to specify a RuntimeType of `List<T>` or `Dictionary<T>` and avoid this logic here.

RuntimeType originally was used to support polymorphic `object` during serialization, but is also used now to map interfaces to a concrete type (as mentioned earlier in other PR about use for `IDictionary<T>`).